src/core/expression/qgsexpressionutils.cpp

```diff
@@ -153,6 +153,21 @@ QgsMapLayer *QgsExpressionUtils::getMapLayerPrivate( const QVariant &value, cons
   return ml;
 }
 
+QgsCoordinateReferenceSystem QgsExpressionUtils::getCrsValue( const QVariant &value, QgsExpression *parent )
+{
+  const bool isCrs = value.userType() == qMetaTypeId<QgsCoordinateReferenceSystem>();
```

Comment:
I think we'll need:
```suggestion
  if ( QgsVariantUtils::isNull( value ) )
    return QgsCoordinateReferenceSystem();
    
  const bool isCrs = value.userType() == qMetaTypeId<QgsCoordinateReferenceSystem>();
```

and maybe also treat empty strings as invalid QgsCoordinateReferenceSystem, instead of setting the evaluation error.

Otherwise eg the logic in the bearing function to fallback to the layer crs won't work, as we'll have set an evaluation error here.

Reply:
Should be fixed, thanks